src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -39,58 +39,80 @@ class TaskReplaceActor(
   // Killed resident tasks are not expunged from the instances list. Ignore
   // them. LaunchQueue takes care of launching instances against reservations
   // first
-  val currentRunningInstances = instanceTracker.specInstancesSync(runSpec.id).filter(_.isActive)
+  var currentRunningInstances: Seq[Instance] = _
```

Comment:
What do you think about removing this mutable state by using a receive method with parameters?

Reply:
It used to be like this in the instance tracker and I did not like it. In this case it's probably not much of a difference since we won't change the code but I find it easier to track this way.